rockcraft/models/project.py

```diff
@@ -219,6 +222,28 @@ def effective_base(self) -> bases.BaseName:
     def model_reference_slug(cls) -> str | None:
         return "/reference/rockcraft.yaml"
 
+    @override
+    def get_build_plan(self) -> list[BuildInfo]:
+        platforms = typing.cast(
+            # https://github.com/canonical/craft-platforms/issues/43
+            craft_platforms.Platforms,  # pyright: ignore[reportPrivateImportUsage]
+            {name: platform.marshal() for name, platform in self.platforms.items()},
+        )
+        build_infos = rock.get_rock_build_plan(
+            base=self.base,
+            build_base=self.build_base,
+            platforms=platforms,
+        )
+        return [
+            BuildInfo(
+                platform=info.platform,
+                build_on=str(info.build_on),
```

Comment:
If we update craft-application's `build-on` and `build-for` to `str | DebianArchitecture`, we can drop this, right?

Reply:
Either that, or we could have a function in craft-application that converts the buildinfos

Reply:
My WIP craft-application change will obviate this with a `from_platforms()` constructor on its BuildInfo. If we want these QoL changes quickly for building the rockcraft/snapcraft versions on I could chop that out into its own PR and you can test from edge?

(The other thing it does is marks `craft_application.BuildInfo` as deprecated, pointing to `craft_platforms`, but that wouldn't make it into this PR since craft-application still uses its own `BuildInfo` everywhere and that'll be a breaking change.)

Reply:
> If we want these QoL changes quickly

Nah I don't think that's necessary, we can take it slow